src/dmd/lexer.d

```diff
@@ -983,6 +996,11 @@ class Lexer
                 return;
             case '#':
                 {
+                    // https://issues.dlang.org/show_bug.cgi?id=22825
+                    // Special token sequences are terminated by newlines,
+                    // and should not be skipped over.
+                    this.tokenizeNewlines = true;
+                    scope(exit) this.tokenizeNewlines = false;
```

Comment:
Instead of a scope(exit), tokenizeNewlines can be set to false on lines 265 and 256.

Everything else looks much better.

Reply:
There's a third place it would need to be unset too. Others may be discovered as well. For instance I haven't tried the following yet:

```
# "surprise
string"
```

Reply:
@WalterBright would inSpecialTokenSequence be a clearer name?

I'm not really a fan of having anyone other than the `case #` setter unsettling this particular boolean field.

Reply:
I don't think a newline in a string literal would cause any problems.

Reply:
_tokenizeNewLines_ seems perfect to me.

As for unsetting tokenizeNewlines, it needs to be unset when a newline token is produced. Conflating this with `scope(exit)` is asking for trouble, as it is not a local scope issue.

Reply:
poundLine can be further simplified to call `scan()` - removing duplication, and giving better error messages for unterminated strings - so unsetting this field in half a dozen more places where endOfLine() is called seems brittle to me.

I'll have a think about what could be done better here to better accommodate that.